Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.swift

```diff
@@ -0,0 +1,164 @@
+                    let attributes = (placement.geoElement as! DynamicEntityObservation).attributes
+                    VStack(alignment: .leading) {
+                        // Display all the attributes in the callout.
+                        ForEach(attributes.sorted(by: { $0.key < $1.key }), id: \.key) { item in
+                            Text("\(item.key): \(String(describing: item.value))")
+                        }
+                    }
+                }
+                .task(id: tappedScreenPoint) {
+                    guard let tappedScreenPoint,
+                          let identifyResult = try? await proxy.identify(
+                            on: dynamicEntityLayer,
+                            screenPoint: tappedScreenPoint,
+                            tolerance: 2
+                          ) else {
+                        // Hides the callout.
+                        calloutPlacement = nil
+                        
+                        return
+                    }
+                    
+                    // Set the callout placement to the observation that was tapped on.
+                    calloutPlacement = identifyResult.geoElements.first.map { .geoElement($0) }
+                }
+        }
+    }
+}
+
+/// The vessel feed that is emitting custom dynamic entity events.
+private struct VesselFeed: CustomDynamicEntityFeed {
+    let events: AsyncThrowingStream<CustomDynamicEntityFeedEvent, Error> = .init { continuation in
+        Task.detached {
+            do {
+                let fileHandle = try FileHandle(forReadingFrom: .selectedVesselsDataSource)
+                let decoder = JSONDecoder()
+                
+                // Loop through each line in the JSON file.
+                for try await line in fileHandle.bytes.lines {
+                    // Delay observations to simulate live data.
+                    try await Task.sleep(nanoseconds: 10_000_000)
+                    
+                    let decodable = try decoder.decode(
+                        AddCustomDynamicEntityDataSourceView.Vessel.self,
+                        from: line.data(using: .utf8)!
+                    )
+                    
+                    // The geometry that was decoded from the JSON.
+                    let geometry = decodable.geometry
+                    
+                    // We successfully decoded the vessel JSON so we should
+                    // add that vessel as a new observation.
+                    continuation.yield(.newObservation(
+                        geometry: Point(x: geometry.x, y: geometry.y, spatialReference: .wgs84),
+                        attributes: decodable.attributes
+                    ))
+                }
+                
+                continuation.finish()
+            } catch {
+                continuation.finish(throwing: error)
+            }
+        }
+    }
```

Comment:
This is much more complicated than it needs to be.

```suggestion
    let events = URL.selectedVesselsDataSource.lines.map { line in
        // Delay observations to simulate live data.
        try await Task.sleep(nanoseconds: 10_000_000)
        
        let decoder = JSONDecoder()
        let vessel = try decoder.decode(
            AddCustomDynamicEntityDataSourceView.Vessel.self,
            from: line.data(using: .utf8)!
        )
        
        // The geometry that was decoded from the JSON.
        let geometry = vessel.geometry
        
        // We successfully decoded the vessel JSON so we should
        // add that vessel as a new observation.
        return CustomDynamicEntityFeedEvent.newObservation(
            geometry: Point(x: geometry.x, y: geometry.y, spatialReference: .wgs84),
            attributes: vessel.attributes
        )
    }
```

Reply:
In this code, we are re-creating the decoder each time the map closure is called, that doesn't seem necessary. Thoughts?

Reply:
I like this implementation of the events async sequence much more though.

Reply:
> we are re-creating the decoder each time the map closure is called

That is intentional and appropriate, as I said in my comment [here](https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/243#discussion_r1296174957). See [this](https://forums.swift.org/t/is-it-ok-to-create-jsondecoder-only-once-in-withthrowingtaskgroup/57260/3) forum post for more information.

Reply:
Great info, thank you.